RUNTIME.md

```diff
@@ -0,0 +1,536 @@
+minimizing energy consumption with marginal degradation of application
+performance.  The root of the control hierarchy tree can communicate
+with the system resource manager to extend the hierarchy above the
+individual MPI application and enable the management of system power
+resources for multiple MPI jobs and multiple users by the system
+resource manager.
+
+The GEOPM package provides two libraries: libgeopm for use with MPI
+applications, and libgeopmpolicy for use with applications that do not
+link to MPI.  There are several command line tools included in GEOPM
+which have dedicated manual pages.  The geopmlaunch(1) command line
+tool is used to launch an MPI application while enabling the GEOPM
+runtime to create a GEOPM Controller thread on each compute node.  The
+Controller loads plugins and executes the Agent algorithm to control
+the compute application.  The geopmlaunch(1) command is part of the
+geopmpy python package that is included in the GEOPM installation.
+See the GEOPM overview man page for further documentation and links:
+[geopm(7)](https://geopm.github.io/man/geopm.7.html).
+
+The GEOPM runtime is extended through three plugin classes: Agent,
+IOGroup, and Comm.  New implementations of these classes can be
+dynamically loaded at runtime by the GEOPM Controller.  The Agent
+class defines which data are collected, how control decisions are
+made, and what messages are communicated between Agents in the tree
+hierarchy.  The reading of data and writing of controls from within a
+compute node is abstracted from the Agent through the PlatformIO
+interface.  This interface provides access to the IOGroup
+implementations that provide a variety of signals and controls.
+IOGroup plugins can be developed independently of the Agents to extend
+the read and write capabilities provided by GEOPM.  The PlatformIO
+abstraction enables Agent implementations to be ported to different
+hardware platforms without modification.  Messaging between Agents
+running on different compute nodes is encapsulated in the Comm class.
+New implementations of the Comm class make it possible to port
+inter-node communication used by the GEOPM runtime to different
+underlying communication protocols and hardware without modifying the
+Agent implementations.
+
+The libgeopm library can be called directly or indirectly within MPI
+applications to enable application feedback for informing the control
+decisions.  The indirect calls are facilitated by GEOPM's integration
+with MPI and OpenMP through their profiling decorators, and the direct
+calls are made through the geopm_prof_c(3) or geopm_fortran(3)
+interfaces.  Marking up a compute application with profiling
+information through these interfaces can enable better integration of
+the GEOPM runtime with the compute application and more precise
+control.
+
+INSTALL
+-------
+The OpenHPC project provides the most robust way to install GEOPM.
```

Comment:
The version in OpenHPC does not have any of the service bits.  Not sure this statement is true anymore.